templates/server/credentialcomposer/credentialcomposer.go

```diff
@@ -0,0 +1,197 @@
+package credentialcomposer
+
+import (
+	"context"
+	"sync"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire-plugin-sdk/pluginmain"
+	"github.com/spiffe/spire-plugin-sdk/pluginsdk"
+	credentialcomposerv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/server/credentialcomposer/v1"
+	configv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/service/common/config/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	// This compile time assertion ensures the plugin conforms properly to the
+	// pluginsdk.NeedsLogger interface.
+	// TODO: Remove if the plugin does not need the logger.
+	_ pluginsdk.NeedsLogger = (*Plugin)(nil)
+
+	// This compile time assertion ensures the plugin conforms properly to the
+	// pluginsdk.NeedsHostServices interface.
+	// TODO: Remove if the plugin does not need host services.
+	_ pluginsdk.NeedsHostServices = (*Plugin)(nil)
+)
+
+// Config defines the configuration for the plugin.
+// TODO: Add relevant configurables or remove if no configuration is required.
+type Config struct {
+}
+
+// Plugin implements the KeyManager plugin
+type Plugin struct {
+	// UnimplementedCredentialComposerServer is embedded to satisfy gRPC
+	credentialcomposerv1.UnimplementedCredentialComposerServer
+
+	// UnimplementedConfigServer is embedded to satisfy gRPC
+	// TODO: Remove if this plugin does not require configuration
+	configv1.UnimplementedConfigServer
+
+	// Configuration should be set atomically
+	// TODO: Remove if this plugin does not require configuration
+	configMtx sync.RWMutex
+	config    *Config
+
+	// The logger received from the framework via the SetLogger method
+	// TODO: Remove if this plugin does not need the logger.
+	logger hclog.Logger
+}
+
+// SetLogger is called by the framework when the plugin is loaded and provides
+// the plugin with a logger wired up to SPIRE's logging facilities.
+// TODO: Remove if the plugin does not need the logger.
+func (p *Plugin) SetLogger(logger hclog.Logger) {
+	p.logger = logger
+}
+
+// BrokerHostServices is called by the framework when the plugin is loaded to
+// give the plugin a chance to obtain clients to SPIRE host services.
+// TODO: Remove if the plugin does not need host services.
+func (p *Plugin) BrokerHostServices(broker pluginsdk.ServiceBroker) error {
+	// TODO: Use the broker to obtain host service clients
+	return nil
+}
+
+// GenerateKey implements the KeyManager GenerateKey RPC
+//func (p *Plugin) GenerateKey(ctx context.Context, req *keymanagerv1.GenerateKeyRequest) (*keymanagerv1.GenerateKeyResponse, error) {
+//	config, err := p.getConfig()
+//	if err != nil {
+//		return nil, err
+//	}
+//
+//	// TODO: Implement the RPC behavior. The following line silences compiler
+//	// warnings and can be removed once the configuration is referenced by the
+//	// implementation.
+//	config = config
+//
+//	return nil, status.Error(codes.Unimplemented, "not implemented")
+//}
+
```

Comment:
do you forgot to remove this?

Reply:
Sure did! Fixed :) Also fixed one other place where I had copied over things from the KeyManager template.